base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_cvd.cc

```diff
@@ -351,12 +351,31 @@ Result<void> SubstituteWithFlag(const std::string& target_dir, const std::vector
   return {};
 }
 
+static constexpr std::string_view kRunfilePrefix = "runfile:";
+
 Result<void> SubstituteWithMarker(const std::string& target_dir, const std::string& marker_file) {
   std::string content;
   CF_EXPECTF(android::base::ReadFileToString(marker_file, &content),
              "failed to read '{}'", marker_file);
   fetch::HostPkgMigrationConfig config;
   CF_EXPECT(google::protobuf::TextFormat::ParseFromString(content, &config), "failed parsing debian_substitution_marker file");
+  std::unique_ptr<Runfiles> runfiles = NULL;
+  auto resolve_runfile  = [&runfiles](const std::string& src) -> Result<std::string> {
```

Comment:
Can this be a couple of functions instead of a lambda? One function to create the runfiles pointer and another to resolve a file given that pointer.